### PR TITLE
Update rubocop and fix errors

### PR DESCRIPTION
**Pull request by bstewart00**
### Comments
CircleCI was hiding the existence of rubocop errors since it always used a cached older version.
Remove unused code (text_caches)

Most errors autofixed `bundle exec rubocop --autocorrect-all`.

I added a ruby 2.5 version requirement on the gem since the autocorrect started using some ruby 2.5 methods. Even 2.5 is a pretty old version at this point so I think it is OK.

### CODE REVIEW

File: lib/wovnrb/headers.rb [hunk 1]
```diff
@@ -115,11 +110,11 @@ def redirect_location(lang)
...
         when 'query'
           lang_param_name = @settings['lang_param_name']
-          if location !~ /\?/
-            location = "#{location}?#{lang_param_name}=#{lang_code}"
-          else @env['REQUEST_URI'] !~ /(\?|&)#{lang_param_name}=/
-               location = "#{location}&#{lang_param_name}=#{lang_code}"
-          end
+          location = if location =~ /\?/
+                       "#{location}&#{lang_param_name}=#{lang_code}"
+                     else
```
Comment (bstewart00) [inline lib/wovnrb/headers.rb:115]: This code is just trying to append the query with either `?` or `#`. The `@env['REQUEST_URI'] !~ /(\?|&)#{lang_param_name}=/` expression was unused.

File: wovnrb.gemspec [hunk 2]
```diff
@@ -11,6 +11,7 @@ Gem::Specification.new do |spec|
   spec.description   = 'Ruby gem for WOVN backend on Rack.'
   spec.homepage      = 'https://wovn.io'
   spec.license       = 'MIT'
+  spec.required_ruby_version = '>= 2.5'
```
Comment (bstewart00) [inline wovnrb.gemspec:14]: I arbitrarily decided 2.5 was a good enough required version.

**Review by zeyuwu — APPROVED** [review]
Looks good.

**Review by Soichiro725 — APPROVED** [review]
I didn't know about the followings.
- lib/wovnrb/text_caches/cache_base.rb
- lib/wovnrb/text_caches/memory_cache.rb

Looks good.

**Pull request merged**